README.md

```diff
@@ -100,4 +100,100 @@ A reducer which returns `null` as the initial state.
 
 ## Example of how to use `extendReducer()` to write your own higher-order reducers.
 
-TBD
+You can look at ready-to-use higher-order reducers, which has been implemented using
+`extendReducer()` helper:
+- [redux-payload](https://github.com/CyberInt/redux-payload)
+- [redux-reset-reducer](https://github.com/CyberInt/redux-reset-reducer)
+
+And start organizing your reducers code in reusable pieces like in following example.
+
+Lets imagine that we have two separate counters and they should provide different behaviours
+on each tick that occurs. One have should only increment its own by 1 and another one should
+also change sign of a number it holds. And you have two separate widgets with such a counters,
+which works independently and starts with different numbers.
+
+```js
+import { compose, combineReducers } from 'redux';
+import { extendReducer, initialReducer } from 'redux-reuse';
+import { ignoreActions } from 'redux-ignore';
+
+const increment = (actionType) => (reducer) =>
+  extendReducer(reducer, {
+    [actionType]: (state) => state + 1
+  });
+
+const changeSign = (actionType) => (reducer) =>
+  extendReducer(reducer, {
+    [actionType]: (state) => state * (-1)
+  });
+
+const widgetReducer = (tickActionType, startingNumber) =>
+  combineReducers({
+    smoothCounter: increment(tickActionType)(initialReducer(startingNumber)),
+    jumpingCounter: compose(
+      increment(tickActionType),
+      changeSign(tickActionType)
+    )(initialReducer(startingNumber))
+  });
+
+const app = combineReducers({
+  widget1: widgetReducer(WIDGET1_TICK, 0),
+  widget2: widgetReducer(WIDGET2_TICK, 100)
+})
+
+```
+
+As you can see everything is reused! :)
+
+Another interesting example is when you need to plug-in and plug-out some functionality. Lets
+imaging we have counter with checkboxes, indicationg what exact functionality should be turned on.
+
+```js
+
+// Modes
+const INCREMENT = 'INCREMENT';
+const SQUARE = 'SQUARE';
+const CHANGE_SIGN = 'CHANGE_SIGN';
+
+// Higher-order reducers
+const increment = (actionType) => (reducer) =>
+  extendReducer(reducer, {
+    [actionType]: (state) => state + 1
+  });
+
+const square = (actionType) => (reducer) =>
+  extendReducer(reducer, {
+    [actionType]: (state) => state * state
+  });
+
+const changeSign = (actionType) => (reducer) =>
+  extendReducer(reducer, {
+    [actionType]: (state) => state * (-1)
+  });
+
+const modesMap = {
+  [INCREMENT]: increment,
+  [SQUARE]: square,
+  [CHANGE_SIGN]: changeSign,
+};
+
+const counter = (state = { modes: {}, value: null }, { payload: mode }) => {
+  switch (action.type) {
+    case ADD_MODE:
+      state.modes[mode] = true;
+      break;
+    case REMOVE_MODE:
+      delete state.modes[mode];
```

Comment:
Redux does not require you to use only immutable data structures, so I think yes, it is ok for simplicity